edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/configsources/FileBackupConfigSource.cs

```diff
@@ -32,43 +34,56 @@ public FileBackupConfigSource(string path, IConfigSource underlying, ISerde<Depl
 
         async Task<DeploymentConfigInfo> ReadFromBackup()
         {
+            DeploymentConfigInfo backedUpDeploymentConfigInfo = DeploymentConfigInfo.Empty;
+
             try
             {
-                if (!File.Exists(this.configFilePath))
-                {
-                    Events.BackupFileDoesNotExist(this.configFilePath);
-                }
-                else
-                {
-                    using (await this.sync.LockAsync())
-                    {
-                        string encryptedJson = await DiskFile.ReadAllAsync(this.configFilePath);
-                        string json = await this.encryptionProvider.DecryptAsync(encryptedJson);
-                        DeploymentConfigInfo deploymentConfigInfo = this.serde.Deserialize(json);
-                        Events.ObtainedDeploymentFromBackup(this.configFilePath);
-                        return deploymentConfigInfo;
-                    }
-                }
+                backedUpDeploymentConfigInfo = await this.lastBackedUpConfig
```

Comment:
The main change here is to cache the last backed up config and to check if it has changed (by implementing IEquatable)?

Reply:
Yes, that is correct.